cli/flox/src/commands/environment.rs

```diff
@@ -1437,6 +1502,68 @@ impl Pull {
     }
 }
 
+fn query_add_system(system: &str) -> bool {
+    let message = format!(
+        "The environment you are trying to pull is not compatible with your system ({system})."
+    );
+    let help = "Use 'flox pull --add-system' to automatically add your system to the list of compatible systems";
+    let confirm_choice =
+        format!("Pull this environment anyway and add '{system}' to the supported systems list.");
+
+    if !Dialog::can_prompt() {
+        return false;
+    }
+
+    let dialog = Dialog {
+        message: &message,
+        help_message: Some(help),
+        typed: Select {
+            options: ["Don't pull this environment", &confirm_choice].to_vec(),
```

Comment:
(contrary to the ticket but I'm guessing that was accidental)
```suggestion
            options: ["Don't pull this environment.", &confirm_choice].to_vec(),
```